### DISCUSSION TITLE
Source control diff keyboard shortcut

### DISCUSSION THREAD

**Discussion by codeyoulateralligator** [microsoft/vscode-discussions#1148] [Extension Development QnA] [answered]
Hi,

![image](https://github.com/microsoft/vscode-discussions/assets/61871309/1881c617-c542-4529-a1e6-4be4c3796b1f)

How to assign a keyboard shortcut to the arrow key that reverts the changes in the source control diff editor?

I don't want to click it with mouse, it is very slow. When i review my changes before commiting i want to go over them and see if i need them all. In this case i have added a comment during debugging which i want to remove. Moving up and down between the changes with keyboard shortcuts in the diff viewer works.

I want to assign a keyboard shortcut like "alt-right" for the arrow key.

**Comment by theReynald** [comment]
Hi @codeyoulateralligator.

Few things to try/consider...

### Steps to Set Up a Keyboard Shortcut in VS Code:

1. **Open Keyboard Shortcuts Editor:**
   - You can open the Keyboard Shortcuts editor by pressing `Ctrl+K Ctrl+S` or by going to `File > Preferences > Keyboard Shortcuts`.

2. **Search for the Relevant Command:**
   - In the Keyboard Shortcuts editor, you can search for commands related to the source control or diff editor. However, the specific command you're looking for, which directly reverts changes from the diff editor, might not be readily visible or named explicitly as such.

3. **Assign a Shortcut:**
   - If you find the command, such as "Revert Selected Ranges" or a similar command applicable in the diff viewer, click on the plus icon (`+`) next to it to assign a new shortcut.
   - Enter your desired key combination, such as `Alt+Right`.

4. **Handling Conflicts:**
   - If the keybinding is already assigned to another command, VS Code will notify you. You can choose to remove the existing binding and assign it to your new command or pick another combination.

### What if the Command Isn't Listed?

If the command to revert changes in the diff editor isn’t available directly through the Keyboard Shortcuts editor, you might need to use an extension that enhances source control capabilities or manually configure commands via extensions that allow custom commands.

### Alternative Approach: Using an Extension
If VS Code does not support this functionality natively in a way you can easily access, consider looking into extensions that enhance the Git experience or allow for more customizable command setups. Extensions like "GitLens" can provide more robust tools for handling diffs and might offer more customizable shortcut options.

Hope this helps.

**Comment by abrilurena** [comment]
Hi, @codeyoulateralligator! do you still need help with this?

**Comment by codeyoulateralligator** [comment]
Yes!

I cannot get it to work:
![image](https://github.com/microsoft/vscode-discussions/assets/61871309/62947f4a-395e-4652-96a7-1116a4ca9048)


Which command do i need to bind for the arrow? 
![image](https://github.com/microsoft/vscode-discussions/assets/61871309/1512df6b-d2fd-424f-ba03-c5849c8d7b10)


In addition it should work more seamlessly, like WinMerge. There you can pull changes from right or push changes from right to left with simple key combinations. This click on arrow button approach is simply not good.

**Reply by abrilurena** [reply]
Hi! i'm adding @isidorn here, he might be able to help ✨

**Reply by isidorn** [reply, marked as answer]
Good question. But this arrow currently does not seem to trigger a globally registered command (at least I do not see it in the output). So I do not think it can be bound.
The best would be to file a new feature request in https://github.com/microsoft/vscode/issues

fyi @hediet in case I missed something

**Comment by abrilurena** [comment]
Hi, @codeyoulateralligator! I hope you're doing great. It's been a while since last time, so I'll be closing this issue for now. Please feel free to re-open if you still need more help about this issue ✨

**Comment by codeyoulateralligator** [comment]
Hi @abrilurena, i am doing good, thanks.

I was waiting for a possible resolution to this issue. 

The issue is definitely NOT resolved (as you can see from the dev chat above).

Please do not mark this issue as "Resolved"

**Reply by robconery** [reply]
Hi @codeyoulateralligator - it sounds as though the command you're using can't be bound, see @isidorn's reply above. If you wouldn't mind, could you add a feature request here? https://github.com/microsoft/vscode/issues.

**Comment by abrilurena** [comment]
Hi, @codeyoulateralligator! Thank you for clarifying the situation and I apologize for any inconvenience. I'll be reopening this issue and adding @theReynald here - maybe he can help ✨
